File: db-sqlAlchemy-alembic/app/models/eintrag.py
# app/models/eintrag.py
from sqlalchemy import Column, Integer, String, Text, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.database import Base

class Eintrag(Base):
    __tablename__ = "eintraege"

    id           = Column(Integer, primary_key=True, index=True)
    titel        = Column(String, nullable=False)
    beschreibung = Column(Text, nullable=True)

    # z. B. Messwert, Status-Code, Bild-URL usw.
    wert         = Column(String, nullable=True)

    erstellt_am  = Column(DateTime(timezone=True), server_default=func.now())

    # FK zum Bericht
    bericht_id   = Column(Integer, ForeignKey("berichte.id", ondelete="CASCADE"), nullable=False)
    bericht      = relationship("Bericht", back_populates="eintraege")

File: kundendokumentation_builder/lib/core/models/eintrag.dart
class Eintrag {
  final int id;
  final String titel;
  final String? beschreibung;
  final String? wert;
  final int berichtId;
  final bool? hasImage;

  Eintrag({
    required this.id,
    required this.titel,
    this.beschreibung,
    this.wert,
    required this.berichtId,
    this.hasImage,
  });

  factory Eintrag.fromJson(Map<String, dynamic> json) {
    return Eintrag(
      id: json['id'],
      titel: json['titel'],
      beschreibung: json['beschreibung'],
      wert: json['wert'],
      berichtId: json['bericht_id'],
      hasImage: json['has_image'] ?? false,
    );
  }
}
